src/com/sap/piper/integration/TransportManagementService.groovy

```diff
@@ -0,0 +1,136 @@
+package com.sap.piper.integration
+
+import com.sap.piper.JsonUtils
+
+class TransportManagementService implements Serializable {
+
+    final Script script
+    final Map config
+
+    def jsonUtils = new JsonUtils()
+
+    TransportManagementService(Script script, Map config) {
+
+        this.script = script
+        this.config = config
+
+    }
+
+    def authentication(String uaaUrl, String oauthClientId, String oauthClientSecret) {
+
+        script.echo "[TransportManagementService] OAuth Token retrieval started."
+
+        if (config.verbose) {
+
+            script.echo "[TransportManagementService] UAA-URL: '${uaaUrl}', ClientId: '${oauthClientId}'"
+
+        }
+
+        def httpResponse = script.sh returnStdout: true,
+            script: """#!/bin/sh -e
+                        curl -XPOST -u '${oauthClientId}':'${oauthClientSecret}' -o responseAuth.txt --write-out '%{http_code}' '${uaaUrl}/oauth/token/?grant_type=client_credentials&response_type=token'
+                    """
+
+        def response = script.readFile("responseAuth.txt")
+
+        if (httpResponse.toInteger() < 200 || httpResponse.toInteger() >= 300) {
+
+            script.error "[TransportManagementService] Retrieval of OAuth-Token failed. HTTP-Status: ${httpResponse} \n [ERROR] Response: '${response}'"
+
+        }
+
+        if (config.verbose) {
+
```

Comment:
nit: removing some empty lines would make the review easier (for me)..

Reply:
Done.